### PR TITLE
docs: add new providers

### PR DESCRIPTION
**Pull request by stefanicjuraj**
Fixes https://github.com/daytonaio/docs/issues/122
Fixes https://github.com/daytonaio/docs/issues/127

Since there are now multiple providers available, the page can be restructured to:

`/configuration/providers#docker`
`/configuration/providers#aws`
`/configuration/providers#digitalocean`
`/configuration/providers#flyio`

Currently, these are available under `/configuration/providers#officially-supported-providers`.

### CODE REVIEW

**Review by ivan-burazin — CHANGES_REQUESTED** [review]
@stefanicjuraj suggest like git providers we have them all on the right side and create a similar flow to the configuration/git-providers page.

Also as mentioned earlier would move targets to their own section

**Comment by stefanicjuraj** [comment]
Modified the provider installation process flow as the newly added providers require setting a target before installing a provider. 

This can later be modified so the target addition process is within the targets section, and it can link the user there to set a target before proceeding with installing a provider.

File: src/content/docs/configuration/providers.mdx [hunk 1]
```diff
@@ -251,33 +145,193 @@ When following the [Set a Target procedure](#set-a-target), note the following c
...
     If you are using password authentication, you can choose "None".
 
-##### Local
+#### Local
 
 By default, Daytona comes pre-configured with a local Docker Target.
-If you have removed the default Docker Target, you can recreate it using the [Set a Target procedure](#set-a-target).
+If you have removed the default Docker Target, you can recreate it using the [set a Target](/configuration/targets#set-a-target) procedure.
+
+## AWS
```
Comment (ivan-burazin) [inline src/content/docs/configuration/providers.mdx:187]: @stefanicjuraj seems we got our wires crossed. This part for AWS is correct but it should be in the Target section.

Here it should be step by step how to install AWS provider and configure it, that part is missing. Please reach out to @Tpuljak for more info and a account for all providers if needed

**Review by ivan-burazin — CHANGES_REQUESTED** [review]

File: src/content/docs/configuration/providers.mdx (hunk 1 above)
Comment (ivan-burazin) [reply src/content/docs/configuration/providers.mdx:187]: @stefanicjuraj edit apologies configuring is target

**Review by ivan-burazin — APPROVED** [review]

**Comment by ivan-burazin** [comment]
Much better but would like a more in depth guide. AI example that gets to the point but needs polish:

Step-by-Step Guide
1. Open Your Terminal
Start by opening your terminal or command prompt where Daytona is accessible.

2. Set the AWS Target
Run the following command to initiate the target setup process:

bash
Copy code
daytona target set
3. Select the AWS Provider
You will be prompted to choose a provider. Select aws-provider from the list:

markdown
Copy code
Choose a provider
1 item
===
aws-provider
v0.0.0
Use the arrow keys or enter the corresponding number to select aws-provider, then press Enter.

4. Create a New Target
After selecting the provider, choose to create a new Target:

sql
Copy code
Select New Target.
5. Enter a Name for Your Target
Provide a unique and descriptive name for your AWS Target. For example:

css
Copy code
Enter a name for your Target.
> my-aws-target
6. Configure AWS Target Options
You will be prompted to enter several configuration options specific to AWS. Below are the options along with explanations and recommendations:

Instance Type

Description: Specifies the type of EC2 instance to launch.
Default: t2.micro
Recommendation: Choose an instance type based on your workload. For development purposes, t2.micro or t3.medium are commonly used.
Input Prompt:
bash
Copy code
The type of instance to launch. Default is t2.micro.
List of available instance types https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html
>
Region

Description: The AWS region where your resources will be hosted.
Default: us-east-1
Recommendation: Choose a region closest to your location or your user base to reduce latency.
Input Prompt:
sql
Copy code
Region
The geographic area where AWS resources are hosted. Default is us-east-1.
>
Secret Access Key

Description: AWS Secret Access Key for authentication.
Note: If left empty, Daytona will fetch it from the AWS_SECRET_ACCESS_KEY environment variable.
Recommendation: It's more secure to use environment variables or AWS IAM roles.
Input Prompt:
mathematica
Copy code
Secret Access Key
If empty, it will be fetched from the AWS_SECRET_ACCESS_KEY environment variable.
>
Volume Size

Description: The size of the EBS volume attached to the instance, in GB.
Default: 20 GB
Recommendation: Ensure it's sufficient for your workspace data. For most development environments, 20-30 GB is adequate.
Input Prompt:
vbnet
Copy code
Volume Size
The size of the instance volume, in GB. Default is 20 GB.
It is recommended that the disk size should be more than 20 GB.
>
Volume Type

Description: The type of EBS volume.
Default: gp3
Recommendation: Choose based on performance and cost needs. Refer to [AWS EBS Volume Types](https://docs.aws.amazon.com/ebs/latest/userguide/ebs-volume-types.html) for details.
Input Prompt:
mathematica
Copy code
Volume Type
The type of volume. Default is gp3.
List of available volume types https://docs.aws.amazon.com/ebs/latest/userguide/ebs-volume-types.html
>
Example Configuration:

Suppose you want to set up a Target named dev-aws-target with the following settings:

Instance Type: t3.medium
Region: us-west-2
Volume Size: 30 GB
Volume Type: gp2
Here's how you would input the details:

bash
Copy code
$ daytona target set
Choose a provider
1 item
===
aws-provider
v0.0.0

Select aws-provider

Select New Target.

Enter a name for your Target.
> dev-aws-target

The type of instance to launch. Default is t2.micro.
List of available instance types https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html
> t3.medium

Region
The geographic area where AWS resources are hosted. Default is us-east-1.
> us-west-2

Secret Access Key
If empty, it will be fetched from the AWS_SECRET_ACCESS_KEY environment variable.
> [Leave blank if using environment variables]

Volume Size
The size of the instance volume, in GB. Default is 20 GB.
It is recommended that the disk size should be more than 20 GB.
> 30

Volume Type
The type of volume. Default is gp3.
List of available volume types https://docs.aws.amazon.com/ebs/latest/userguide/ebs-volume-types.html
> gp2

# Press Enter to confirm
After completing the prompts, you should see:

arduino
Copy code
Target set successfully
7. Verify the Target
To ensure that your AWS Target has been set up correctly, list all configured Targets:

bash
Copy code
daytona target list
You should see an output similar to:

mathematica
Copy code
 Target         Provider        Options
─────────────────────────────────────────────────────────────────────────
 dev-aws-target aws-provider    {
                                     "Instance Type": "t3.medium",
                                     "Region": "us-west-2",
                                     "Volume Size": "30",
                                     "Volume Type": "gp2"
                                 }
This confirms that your AWS Target is correctly configured and ready for use.

**Review by ivan-burazin — CHANGES_REQUESTED** [review]
Comment left in conversation

File: src/content/docs/configuration/providers.mdx [hunk 2]
```diff
@@ -94,190 +81,196 @@ Daytona allows you to update an existing Provider, enabling you to apply the lat
...
+    ```
 
-        Remote User
-        Note: non-root user required
-        >
+## Docker
 
-        Sock Path
-        > /var/run/docker.sock
+The Docker Provider allows Daytona to create Workspace projects on Docker containers. The Provider integrates Daytona with Docker, enabling you to manage Workspaces on a Docker instance. Daytona installs a Provider for Docker (`docker-provider`) by default and adds a default [Target](/configuration/targets#docker) using the Docker Provider, allowing the creation of Workspaces on your local machine.
```
Comment (Tpuljak) [inline src/content/docs/configuration/providers.mdx:118]: `The Docker Provider allows Daytona to create Workspace projects on Docker containers. The Provider integrates Daytona with Docker, enabling you to manage Workspaces on a Docker instance.`

=> `The Docker Provider allows Daytona to create Workspace projects as Docker containers`

File: src/content/docs/configuration/providers.mdx [hunk 3]
```diff
@@ -94,190 +81,196 @@ Daytona allows you to update an existing Provider, enabling you to apply the lat
...
+    expanded
+  />
+</DocumentList>
 
-### Docker
+## DigitalOcean
 
-After installation, Daytona installs a Provider for Docker (`daytona-provider-docker`) by default.
-A default Target using the Docker Provider is also added, allowing the creation of Workspaces on your local machine.
+The DigitalOcean Provider allows Daytona to create Workspace projects on DigitalOcean VMs, known as droplets. The Provider integrates Daytona with DigitalOcean, enabling you to manage Workspaces on a flexible and scalable cloud platform.
```
Comment (Tpuljak) [inline src/content/docs/configuration/providers.mdx:200]: `Droplets` (capital D)

File: src/content/docs/configuration/targets.mdx [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+    ```
+
+3. Select `local` to set a local Docker target, or `New Target` to set a remote Docker target.
+
+    - A local machine uses your local installation of Docker to deploy Workspaces and requires configuring the `Sock Path` to the Docker socket.
+
+    - A remote machine uses a remote installation of Docker to deploy Workspaces and requires configuring the remote `password`, `port`, `user`, `sock path`, `Workspace data directory`, and `private key path`.
+
+4. Enter a name for your Target.
+5. Enter the appropriate configuration options when prompted. The configuration options vary based on the selected Provider. The following example shows setting a local Docker Target.
```
Comment (Tpuljak) [inline src/content/docs/configuration/targets.mdx:163]: > The configuration options vary based on the selected Provider

Docker was already selected, but the options vary based on the name of the target actually.

If the name is `local`, then only the `Sock Path` is required, if the name is anything else, then the user is configuring a remote target so the remote options are required

File: src/content/docs/configuration/targets.mdx [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+        ===
+        aws-provider
+        v0.0.0
+    ```
+
+3. Select `New Target`.
+4. Enter a name for your Target.
+5. Enter the appropriate configuration options when prompted.
+
+    ```text
```
Comment (Tpuljak) [inline src/content/docs/configuration/targets.mdx:200]: https://github.com/daytonaio/daytona-provider-aws/issues/3

The target option descriptions will probably be updated before the release so just putting it here for tracking

**Review by Tpuljak — CHANGES_REQUESTED** [review]

**Review by ivan-burazin — APPROVED** [review]

File: src/content/docs/configuration/providers.mdx [hunk 6]
```diff
@@ -94,190 +81,188 @@ Daytona allows you to update an existing Provider, enabling you to apply the lat
...
+    ```
 
-        Remote User
-        Note: non-root user required
-        >
+## Docker
 
-        Sock Path
-        > /var/run/docker.sock
+The Docker Provider allows Daytona to create Workspace projects as Docker containers. The Provider integrates Daytona with Docker, enabling you to manage Workspaces on a Docker instance. Daytona installs a Provider for Docker (`docker-provider`) by default and adds a default [Target](/configuration/targets#docker) using the Docker Provider, allowing the creation of Workspaces on your local machine.
```
Comment (Tpuljak) [inline src/content/docs/configuration/providers.mdx:118]: `The Provider integrates Daytona with Docker, enabling you to manage Workspaces on a Docker instance.`

I think this is redundant and `on a Docker instance` sounds a bit weird.

File: src/content/docs/configuration/targets.mdx [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+        v0.0.0
+        fly-provider
+        v0.0.0
+        docker-provider
+        v0.0.0
+    ```
+
+3. Select `New Target`.
+4. Enter a name for your Target.
+5. Enter the appropriate configuration options when prompted. The configuration options vary based on the selected Target. The following example shows setting a remote Docker Target.
```
Comment (Tpuljak) [inline src/content/docs/configuration/targets.mdx:48]: `based on the selected Target` => `based on the selected Provider`

**Review by Tpuljak — CHANGES_REQUESTED** [review]

**Review by Tpuljak — COMMENTED** [review]
All good now. I'll approve tomorrow once we release the providers officially and update the target descriptions here

**Comment by stefanicjuraj** [comment]
Added a commit https://github.com/daytonaio/docs/pull/124/commits/aadffb6ae67c49f49dc8ee789c4c29f13c7fb000 to cover the new issue.

Fixes https://github.com/daytonaio/docs/issues/127

**Review by ivan-burazin — APPROVED** [review]

**Review by Tpuljak — APPROVED** [review]

**Pull request merged**